### ISSUE TITLE
phpmyadmin Login issue WD myCloud Ex4100 mysqli::real_connect(): (HY000/2002): Connection refused

### ISSUE THREAD

**Issue by Enicma** [phpmyadmin/phpmyadmin#18206]
### Describe the bug

I have a simple Problem with my Database from phpmyadmin on Western Digital myCloud Ex4100 NAS System.
See attached picture.
 
I am using it long time for my Kodi Database to sync the Database for 4 Kodi instances in my home.
Since 4 weeks, Kodi (Kodi release V20.0) can not sync anymore to the Database.
 
Kodi is running without connecting to Database fine.
 
Then I check is it possible to login on my NAS Western Digital myCloud Ex4100 to login in the Database.
But no password was working From all my users.
 
So I decided to uninstall phpmyadmin from my NAS.
And install it new.
 
Then I try all standard Password, what I found.
user: Password
root; root
root
admin
admin; admin
root; admin
root; leer
root; leeres passwort
 
But nothing is working.
What can I do?
Is there a possiblity to delete all existing and start with default.
 
I always get same massage.
mysqli::real_connect(): (HY000/2002): Connection refused
 
I hope you can give me an idea.
 
phpmyadmin
Version1.13 is installed on NAS

**Comment by Enicma** [comment]
![PHPmyadmin issue](https://user-images.githubusercontent.com/38335049/222731645-ae46c8b0-1a26-47dc-a84a-ec4cf8bf18b7.jpg)

**Comment by ibennetch** [comment]
Can you get to a command prompt on your NAS? If so, I'd try using the command line client to connect, try `mysql -h 127.0.0.1 -u root -p` and `mysql -h localhost -u root -p`, or use whichever username you usually have for this service instead of root.

Are you able to see that the MySQL server is actually running correctly? Some sort of process monitor or error log, but I am not super family with this WD software to know how to do that.

The MySQL manual has various [instructions for resetting the root password](https://dev.mysql.com/doc/refman/8.0/en/resetting-permissions.html), if you have sufficient access and need to, but it doesn't really make sense that this would randomly change so I would investigate other options first.

**Comment by Enicma** [comment]
![Nas Space](https://user-images.githubusercontent.com/38335049/222972571-a903c7fb-91f0-40a0-b0ef-147958f89f5f.jpg)

**Comment by Enicma** [comment]
I have no Idea how I can get a command prompt on Western Digital myCloud Ex4100 NAS System.
I found a lot of log files. But no Idea which one is the file where are interesting information inside.
se the Create and Save System Report via the WD My Cloud Dashboard > > > Create and Save System Report > Create and Save.

After the zip file is saved one can extract the file contents and view several of the log files.

**Comment by Enicma** [comment]
How can i found out if MySQL Server / phpmyadmin are running on Western Digital myCloud Ex4100 NAS System.

**Comment by Enicma** [comment]
I just delete phpmyadmin. Switch off the NAS and restart. Then install phpmyadmin again.

I Just found in a log file:
![grafik](https://user-images.githubusercontent.com/38335049/222975067-7067b39f-1b90-43b8-8f0d-5579c41b2cfc.png)

**Comment by liviuconcioiu** [comment]
According to https://support-en.wd.com/app/answers/detailweb/a_id/10790/kw/mysql you have to log in via SSH (https://support-en.wd.com/app/answers/detailweb/a_id/15984) and then run

`python3 /usr/bin/mysql_reset.py -r`

The username for phpMyAdmin should be `root` and no password.

Also, that isn't the phpMyAdmin version (1.13), but a version number that WD gave.

**Comment by williamdes** [comment]
When the error is
> mysqli::real_connect(): (HY000/2002): Connection refused

It's most proenot related to phpMyAdmin. Please check your database server configuration and access controls.
How did you setup the database?

**Comment by Enicma** [comment]
I have downloaded Putty
![grafik](https://user-images.githubusercontent.com/38335049/223228939-22f3ab01-f6d8-43a9-bb2c-8e02a48cb23f.png)
Setup like your manual and connect.
This is the Feedback:
![grafik](https://user-images.githubusercontent.com/38335049/223229150-2ecb28e5-d73f-4dc6-84e3-837dfcafc69a.png)
And now I can Login with the User root and empty password field.
![grafik](https://user-images.githubusercontent.com/38335049/223229661-b2466a2d-653f-40a4-8cf5-ac6914002504.png)

Thanks a lot for your help.!!

One important thing for all Western Digital myCloud Ex4100 NAS System user always check this settings first if you have any issue with Kodi Database:
![grafik](https://user-images.githubusercontent.com/38335049/223230167-08aab702-7e71-40a2-85c5-e0a79217d987.png)
![grafik](https://user-images.githubusercontent.com/38335049/223230226-b6c9e8cf-46ee-4d7a-b5d0-aff9c6792f61.png)
![grafik](https://user-images.githubusercontent.com/38335049/223230290-7a88e403-ed7f-4931-ab58-665240718ff7.png)

Because it change sometimes from alone for example during update!!! I think it is a WD Bug I report to WD as ticket

**Comment by Liese2007** [comment]
Hi,
it seems I have the same issue on a WD myCLoud Ultra Ex2.
I can login via SSH , but cannot reset the root password to EMPTY.

It seems that in between last post on March 5 and 1st post on March 6, there is a hint not shown on HOW TO install mysql on the NAS.
Because @Enicma is writing "...Setup like your manual and connect....."
Did I miss a link to "your manual" ?

Regards
Joerg

**Comment by Enicma** [comment]
Here are some interesting links:
https://support-en.wd.com/app/answers/detailweb/a_id/10790/kw/mysql

https://support-en.wd.com/app/answers/detailweb/a_id/26497/session/L2F2LzEvdGltZS8xNzA0MzI4MzMyL2dlbi8xNzA0MzI4MzMyL3NpZC9mVXRkNWVDTUZSeHZwNVZzeDhfeE5LWEFOTlNQU2lnM2tVN0M3em5LYlNuNGV4dDVTQnVUN3ZnV2hFT3l1QVlvMEFsRmdFSXp0MktUZmo4MFoxeFNVaEEyZ1RON0Mzb0dvclolN0U1TlpHb29PeVRIZVcyS0lvN3RXZyUyMSUyMQ%3D%3D#subject1

https://support-en.wd.com/app/answers/detailweb/a_id/24926/related/1

I hope that helps.

In the moment I have a also Ticket open at WD Support.
After last Firmware update to Version 5.27.157
on every restart the Setting for 
![grafik](https://github.com/phpmyadmin/phpmyadmin/assets/38335049/0f4a6f40-4f83-4165-9345-a538500bd232)
Network Database On with Remote access change back to On without remote access.

I hope WD Support can help me with this issue.

**Comment by Liese2007** [comment]
OK, I did the same as you posted above.
The WD support told me to remove the MySQL database via SSH.
I did not realize , that you posted it above - so once more:

Login via SSH:
How to Access WD My Cloud Using SSH (Secure Shell)
https://support-in.wd.com/app/answers/detailweb/a_id/26497/#subject1

Remove MySQL Database after login as root (sshd on a Ultra Ex2):
rm -rf /mnt/HD_a4/.@database@/*

And then turn off and turn on MySQL in adminUI
(Deutschsprachige Oberfläche: EINSTELLUNGEN --> NETZWERK , Netzwerk-Datenbank AUS und wieder EIN schalten)

New installation of phpmyadmin.
And root login on MySQL Database is OK.
